e2e/e2e_test.go

```diff
@@ -457,6 +439,62 @@ func (suite *TestSuite) TestCanForceRenewalWithSignal() {
 	assert.NotEqual(t, renewedCert.Leaf.SerialNumber, initialCert.Leaf.SerialNumber)
 }
 
+func (suite *TestSuite) runSelfSignedAcmeBuddy(domain string, opts ...tc.ContainerCustomizer) (*tc.DockerContainer, error) {
+	ctx := suite.T().Context()
+
+	opts = append([]tc.ContainerCustomizer{
+		tc.WithLogConsumers(&tc.StdoutLogConsumer{}),
+		network.WithNetwork([]string{}, suite.network),
+		tc.WithCmd(
+			"--self-signed",
+			"--domain", domain,
+			"--email=admin@example.com",
+			"--certificate-path=/tls/certificate.pem",
+			"--key-path=/tls/key.pem",
+		),
+	}, append(opts, WithDefaultVolumeMount("/tls"))...)
+
+	return tc.Run(ctx, suite.image, opts...)
+}
+
+func (suite *TestSuite) TestSelfSignedCertificateGeneration() {
+	t := suite.T()
+	ctx := t.Context()
+
+	c, err := suite.runSelfSignedAcmeBuddy("example.com")
+	if err != nil {
+		t.Fatalf("failed to start AcmeBuddy in self-signed mode: %v", err)
+	}
+	defer c.Terminate(ctx)
+
+	time.Sleep(2 * time.Second)
```

Comment:
I don't really like putting arbitrary Sleeps like that in test code. It makes tests slow and flaky.
I think it would be fine to write this test with `--oneshot`, then it is easy to know when to read the certificate. See the `obtainCertificate` function.